### ISSUE TITLE
After the update to 2.1-preview2 new DocumentClient() fails only on Azure, killing the whole application

### ISSUE THREAD

**Issue by faso** [aspnet/Home#3115]
Posting this here because the DocumentClient library doesn't seem to have a proper separate repo. I've also filed it here: https://github.com/Azure/azure-documentdb-dotnet/issues/495, but it seems pretty dead to me.

I am having a very strange issue here. This all started happening after the update to preview2 of .NET Core 2.1.

The project works totally fine on my local machine, but fails after a few clicks when hosted on Azure. I have isolated the issue to a single line, which is 
`new DocumentClient(settings.Endpoint, settings.ReadKey);`
The settings are fine, the same issue appears if I hardcode the strings. 

This line is called in a constructor for a service that's registered in `Startup.cs`.

The strange thing is that I don't get any exceptions, I don't get any DI errors, after a few minutes it just says that the process was terminated and the whole application stops working. It also happens on exactly the 2nd time I'm calling the controller into which I'm injecting the service.

Any pointers or hints?

I'm using the latest (1.9.1.) version of the `Microsoft.Azure.DocumentDB.Core` assembly and the `dotnet --version` output is `2.1.300-preview2-008530`, both on local and on Azure.

**Comment by Tratcher** [comment]
Can you share a small app that reproduces the issue?

**Comment by faso** [comment]
Okay, so I've resolved the issue. I had to register DocumentClient as a singleton. That, however, does not explain why it was not failing on my local machine (and on local machines of many of my colleagues). An explanation would be fantastic.

It would be great if the rule about using a singleton was documented better and not hidden in a single blog post about performance tips. I will create an issue about this, as I believe this is extremely important in saving a lot of time for people in similar situations.
